ci_framework/roles/rhol_crc/tasks/main.yml

```diff
@@ -0,0 +1,43 @@
+---
+# Copyright Red Hat, Inc.
+# All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License. You may obtain
+# a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+
+- name: Check RHOL/CRC binary exists
+  ansible.builtin.stat:
+    path: "{{ rhol_crc_binary }}"
+  register: rhol_crc_binary_stat
+
+- name: Get RHOL/CRC binary if does not exist
+  ansible.builtin.include_tasks: tasks/binary.yml
+  when: not rhol_crc_binary_stat.stat.exists
+
+- name: Get VM domains
+  community.libvirt.virt:
+    command: list_vms
+  register: vm_domains
+  
+- name: Fail if crc domain is already defined
+  ansible.builtin.fail:
+    msg: "The crc domain already exist. Interrupting tasks..."
+  when: ("'crc' | string in vm_domains.list_vms" and not rhol_crc_force_cleanup|bool)
+
+- name: Set RHOL/CRC configuration options
+  ansible.builtin.include_tasks: tasks/configuration.yml
+
+- name: Setup RHOL/CRC
+  ansible.builtin.shell: "{{ rhol_crc_binary }} setup"
+
+- name: Start RHOL/CRC
+  ansible.builtin.shell: "{{ rhol_crc_binary }} start"
```

Comment:
so afaik here we want to register the output.

Note: most of the `shell` may be a simple `command` - it's safer and better.